src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -494,6 +494,35 @@ def upgrade_stop(self) -> str:
         self.mgr.event.set()
         return 'Stopped upgrade to %s' % target_image
 
+    def upgrade_service(self, service_type: str, service_image: str, image: str) -> List[str]:
+        key = f'mgr/cephadm/container_image_{service_image}'
+        try:
+            self.mgr.check_mon_command({
+                'prefix': 'config set',
+                'name': key,
+                'value': image,
+                'who': 'mgr',
+            })
```

Comment:
we shouldn't need to run a mon command to update config options that are part of the cephadm module. You should just be able to access `self.mgr.<setting>` directly, e.g. `self.mgr.container_image_prometheus = <image-name>`